sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -820,12 +820,27 @@ def request_status(self, status_link: str) -> PipelineResponse[HttpRequestTypeVa
         """
         if self._path_format_arguments:
             status_link = self._client.format_url(status_link, **self._path_format_arguments)
+        request_params: Dict[str, Any] = {}
+        if self._lro_options and "api_version" in self._lro_options:
+            parsed_status_link = urllib.parse.urlparse(status_link)
+            request_params = cast(
+                Dict[str, Any],
+                case_insensitive_dict(
```

Comment:
You don't need the case_insentive_dict, which creates a cast that is unexpected. Since you're looping through the query items anyway, you can just call `key.lower()` and you will be assured that `api-version` can be reached.

Reply:
Make sense!